File: Algorithms/Sorting/selection_sort.js
function selectionSort(array) {
    // Time complexity: O(n^2)
    // Space complexity: O(1)
    for (let trav_index=array.length; trav_index > 0; trav_index--) {
        for (let i=array.length-trav_index+1; i < array.length; i++) {
            if (array[i] < array[array.length-trav_index]) {
                [array[i], array[array.length-trav_index]] = 
                [array[array.length-trav_index], array[i]];
            }
        }
    }
    return array;
}

if (require.main === module) {
    console.log("Sorting [1, 5, 3, 2, 9, 8, 6]:",
    selectionSort([1, 5, 3, 2, 9, 8, 6]));
    console.log("Sorting [7, 2, 1, 9, 3, 2]:",
    selectionSort([7, 2, 1, 9, 3, 2]));
}

File: Algorithms/Sorting/selection_sort.py
def selection_sort(input_list: list):
    # Time complexity: O(n^2)
    # Space complexity: O(1)
    if not (isinstance(input_list, list) and input_list):
        return TypeError("Check your input")
    for traverse_index in range(len(input_list), 0, -1):
        for i in range(len(input_list)-traverse_index+1, len(input_list)):
            if input_list[i] < input_list[len(input_list)-traverse_index]:
                input_list[i], input_list[len(input_list)-traverse_index] = \
                    input_list[len(input_list)-traverse_index], input_list[i]
    return input_list


if __name__ == "__main__":
    print("Sorting [1, 5, 3, 2, 9, 8, 6]:",
    selection_sort([1, 5, 3, 2, 9, 8, 6]))
    print("Sorting [7, 2, 1, 9, 3, 2]:",
    selection_sort([7, 2, 1, 9, 3, 2]))
